### PR TITLE
Shared Library

### PR DESCRIPTION
**Pull request by yujinyuz**
Please ensure you are familiar with our Pull Request Description expectations described here: https://www.pullrequest.com/blog/writing-a-great-pull-request-description/
### What?

#1546 

### Screenshots (if front end is affected)

https://github.com/bytedeck/bytedeck/assets/10972027/4cb16065-2d02-4517-a3b0-56d37f04cd5e

### Review request
@tylerecouture

## Summary by CodeRabbit

- **New Features**
	- Introduced a new Library module in the application, accessible via the `/quests/library/` route.
	- Added functionality for importing quests between different schemas with visibility control.
	- Implemented a new configuration option to enable or disable the Shared Library feature.
	- Added a "Library" tab in the quest management interface, conditional on user roles and settings.

- **Documentation**
	- Placeholder comments added for future model registration and model creation in the Library app.

- **Style**
	- Updated UI to include a new "Library" tab based on configuration and user roles.

- **Tests**
	- Added tests for new Library functionalities, including quest import and schema context setup.

- **Chores**
	- Database migration for adding the `enable_shared_library` setting.

### CODE REVIEW

**Comment by tylerecouture** [comment]
@coderabbitai review

**Comment by yujinyuz** [comment]
@coderabbitai review

**Review by tylerecouture — CHANGES_REQUESTED** [review]
This is wicked!  I haven't had a chance to fully review this yet, but I would like to get it into production even as a Beta so we can test for real.  To do that though, we need **a SiteConfig setting that is off/False by default to turn on the Library.**  That way we can test in out in the wild with all the complexity that real quests and campaigns etc come with.

**Comment by yujinyuz** [comment]
@tylerecouture alright that should be straightforward. I'll try to update the PR ASAP

**Review by tylerecouture — CHANGES_REQUESTED** [review]
An awesome first step I love it!  A few things before we can merge this WIP into develop:

1. For new dev environments, we shouldn't get this when trying to use the library (cus library deck doesn't exist).  So we need to have the library app created when a new dev environment is setup  (can do that in hackerspace_online/management/commands/ initdb.py?)

image

2. Test that the library tab view should not setup a library tenant, that needs to already exist
3. Can you move the siteconfig library toggle into the Advanced section (only changeable by the deck owner), and it the setting's label, make it ` Enable Shared Library - EXPERIMENTAL WIP`
4. When I import a quest that already exists, I get a success message, but it doesn't import.   It SHOULDN"T import, so that's good, but I should get a more informative message, and not green =)  Maybe a warning more like this: "Your deck already has a quest named {{local_quest_name}} with the import_id {{import_id}}.  The quest was not imported.

**Comment by tylerecouture** [comment]
Good point. Maybe we just need to check in the test setup that the library tenant exists already, and if it doesn't, then make it?

File: src/hackerspace_online/management/commands/initdb.py [hunk 1]
```diff
@@ -23,6 +24,25 @@ class Command(BaseCommand):
...
+    def setup_shared_library(self):
+        self.stdout.write('\n** Setting up shared library...')
+        library_tenant, created = Tenant.objects.get_or_create(
+            schema_name='library',
+            name='Shared Library'
+        )
+
+        if not created:
+            library_tenant.domains.create(
+                domain='library.' + settings.ROOT_DOMAIN,
```
[the remark this answers is not in the document]
Comment (yujinyuz) [reply src/hackerspace_online/management/commands/initdb.py:36]: @coderabbitai I don't think it makes sense to use `os.path.join` on urls

**Comment by yujinyuz** [comment]
@tylerecouture addressed request changes

**Comment by tylerecouture** [comment]
@coderabbitai review

**Comment by tylerecouture** [comment]
@coderabbitai review

**Pull request merged**